features/users/operations.py

```diff
@@ -0,0 +1,124 @@
+from starlette.responses import JSONResponse
+
+from db.connection import get_session
+
+import bcrypt
+
+from datetime import datetime, timedelta
+from typing import Union, Any
+from jose import jwt
+
+from .exceptions import password_validation_exception, user_already_exist, access_denied
+from .input_models import RegisterUserInputModel
+from .models import User
+
+import configuration
+
+config = configuration.Config()
+
+
+def successfully_created_user_response(user: User) -> JSONResponse:
+    return JSONResponse(
+        content={
+            "message": "Successfully created",
+            "user_data": {"id": user.id, "username": user.username, "email": user.email}
+        },
+        status_code=201
+    )
+
+
+def hash_password(password: str):
+    salt = bcrypt.gensalt()
+    hashed_password = bcrypt.hashpw(password.encode("utf-8"), salt)
+    return hashed_password
+
+
+def check_password(user: User, password: str) -> bool:
+    # Check if passwords match (for login)
+    return bcrypt.checkpw(password.encode('utf-8'), user.password)
+
+
+def set_password(user: RegisterUserInputModel, password: str):
+    user.password = hash_password(password=password)
+
+
+def create_new_user(user: RegisterUserInputModel) -> User:
+    # Check if the username or email already exists
+    db = get_session()
+    if get_user(username=user.username, email=user.email):
+        # Raise an HTTPException with the error message
+        user_already_exist()
+
+    # Validate password
+    try:
+        RegisterUserInputModel.validate_password(user.password)
+    except ValueError as e:
```

Comment:
This validation is redundant, because the RegisterUserInputModel is validated upon creation